src/Test/ExecutionResult.php

```diff
@@ -0,0 +1,167 @@
+<?php
+
+/*
+ * This file is part of the Fidry\Console package.
+ *
+ * (c) Théo FIDRY <theo.fidry@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Fidry\Console\Test;
+
+use PHPUnit\Framework\Assert;
+use Symfony\Component\Console\Tester\Constraint\CommandIsSuccessful;
+use function array_filter;
+use function call_user_func;
+use function function_exists;
+use function implode;
+use function str_replace;
+use const PHP_EOL;
+
+final class ExecutionResult
+{
+    public static function fromTestOutput(
+        string $intput,
+        int $statusCode,
+        TestOutput $testOutput,
+    ) {
+        return new self(
+            $statusCode,
+            $testOutput->getOutputContents(),
+            $testOutput->getErrorOutputContents(),
+            $testOutput->getDisplayContents(),
+        );
+    }
+
+    public function __construct(
+        private string $input,
+        private int $statusCode,
+        private string $output,
+        private string $errorOutput,
+        private string $display,
+    ) {
+    }
+
+    /**
+     * Gets the status code returned by the execution of the command or application.
+     */
+    public function getStatusCode(): int
+    {
+        return $this->statusCode;
+    }
+
+    /**
+     * Gets the display returned by the execution of the command or application. The display combines what was
+     * written on both the output and error output.
+     *
+     * @param bool $normalize Whether to normalize end of lines to \n or not.
+     */
+    public function getDisplay(bool $normalize = false): string
```

Comment:
part of the proposal, rather than going with `display` and `error display` like done in ZendStruck's ConsoleTest, I choose:

- display: stdout + stderr
- output: stdout
- error output: stderr

(stdout and stderr are the "defaults", as technically you could configure different outputs).